### DISCUSSION TITLE
Some sort of PIN

### DISCUSSION THREAD

**Discussion by VeryUsual** [InfiniTimeOrg/InfiniTime#1732] [Ideas]
Adding a PIN unlock feature to InfiniTime could potentially benefit the community by having enhanced security. A PIN requirement would secure personal data and apps on the smartwatch from unauthorized access. This provides protection if the watch is stolen 

In summary, adding PIN unlock can directly enhance security and privacy - benefiting everyone!

**Comment by srt54558** [comment]
This doesn't make sense, because infinitime doesn't have any personal data. But in the future it may be useful

**Comment by borkymcgee** [comment]
I tend to agree with @srt54558, there really ISN'T anything much to protect right now, the most a malicious actor could do is see undismissed notifications and (god forbid) pause your music.

This is on top of the fact that the pinetime, as a smartwatch, is designed to be strapped to your body almost all of the time.

Adding a PIN would also make use of the watch a little less convenient.

On top of all of this, i can't help but feel like this question was written by an LLM. It's long winded, shows a lack of understanding of the project "like fingerprint scanning if supported by hardware" (it isn't), and the complete change in tone between the first sentence and the rest of the post.

**Comment by JustScott** [comment]
I think this could be helpful if implemented with some kind of core KeyRing service, in which certain apps have at rest encryption, and when you open the app you're first prompted to enter a pin to unlock the keyring, which then decrypts the keys used to encrypt the apps data. I don't think a pin code should be required to get past the current time screen (like apple does), as this is very annoying in my opinion.

No apps exist right now that require encryption/storing secrets, but I would really like to someday have a TOTP 2FA application using the recent keyboard PR: #1466.
